source/blink.cpp

```diff
@@ -133,30 +153,48 @@ void blink::application::run()
 		print("  Started process with PID " + std::to_string(pi.dwProcessId));
 	}
 
-	print("Starting file system watcher for '" + _source_dir.string() + "' ...");
+	size_t dir_index = 0;
+	std::vector<scoped_handle> dir_handles;
+	std::vector<scoped_handle> event_handles;
+	std::vector<notification_info> notification_infos;
+	for (auto it = _source_dirs.begin(); it != _source_dirs.end(); ++it) {
+		auto source_dir = *it;
+		print("Starting file system watcher for '" + source_dir.string() + "' ...");
 
-	// Open handle to the common source code directory
-	const scoped_handle watcher_handle = CreateFileW(_source_dir.native().c_str(), FILE_LIST_DIRECTORY, FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE, nullptr, OPEN_EXISTING, FILE_FLAG_BACKUP_SEMANTICS, nullptr);
+		dir_handles.push_back(INVALID_HANDLE_VALUE);
+		event_handles.push_back(INVALID_HANDLE_VALUE);
+		notification_info info;
+		notification_infos.push_back(info);
 
-	if (watcher_handle == INVALID_HANDLE_VALUE)
-	{
-		print("  Error: Could not open directory handle.");
-		return;
+		if (!set_watch(dir_index, dir_handles, event_handles, notification_infos))
+			return;
+		++dir_index;
 	}
 
-	DWORD size = 0; BYTE watcher_buffer[4096];
+	DWORD size = 0;
+	while (PeekNamedPipe(compiler_stdout, nullptr, 0, nullptr, &size, nullptr)) {
+		const DWORD wait_result = WaitForMultipleObjects(event_handles.size(), &event_handles[0], FALSE, INFINITE);
```

Comment:
Style nitpick: `event_handles.data()` instead of `&event_handles[0]`

Reply:
Thanks. Using .data() now.